### PR TITLE
WIP

### PR DESCRIPTION
**Pull request by sohamganatra**
### **PR Type**
enhancement, documentation

___

### **Description**
- Added `Type` import to `action.py`.
- Improved field descriptions and docstrings in various cmd_manager actions.
- Commented out `SetCursors` action in `cmd_manager_tool.py`.
- Removed `command_runner_config.py` file.
- Updated docstrings for various classes to simplify and enhance clarity.
- Refactored `try-swe.py` for improved readability and functionality.
- Simplified and updated task configuration in `task_config.yaml`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>action.py</strong><dd><code>Added `Type` import to action.py.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/action.py

- Added `Type` to the import statement from `typing`.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-6f9cf1fc507fd9e4c13cfca1ffd2ca9b9f0d586702691c605086064bcdfb8c20">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>cmds.py</strong><dd><code>Improved field descriptions and docstrings in cmd_manager actions.</code></dd></summary>
<hr>
      
composio/local_tools/local_workspace/cmd_manager/actions/cmds.py

<li>Updated descriptions for <code>line_number</code> and <code>file_name</code> fields.<br> <li> Simplified <code>_display_name</code> attributes for various classes.<br> <li> Enhanced docstrings for <code>CreateFileCmd</code> and <code>EditFile</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-4d5c577004c50cee89b184d140d51a9cf8203dcc49a8a23619d7dcc6cdff03cb">+7/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>edit_cmd.py</strong><dd><code>Improved field descriptions and docstrings in edit_cmd actions.</code></dd></summary>
<hr>
      
composio/local_tools/local_workspace/cmd_manager/actions/edit_cmd.py

<li>Updated descriptions for <code>start_line</code>, <code>end_line</code>, and <code>replacement_text</code> <br>fields.<br> <li> Simplified <code>_display_name</code> attribute.<br> <li> Enhanced docstring for <code>EditFile</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-d9e8573a9ae3232fa1611ffb7c91d657303bac3775def78d24d00767d494d60e">+13/-10</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>run_cmd.py</strong><dd><code>Enhanced docstring and simplified display name in run_cmd actions.</code></dd></summary>
<hr>
      
composio/local_tools/local_workspace/cmd_manager/actions/run_cmd.py

<li>Enhanced docstring for <code>RunCommandOnWorkspace</code>.<br> <li> Simplified <code>_display_name</code> attribute.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-249ef5e2c06694510a700363efd38497f39549c65557eda8b15a43f3f675fc11">+11/-2</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>scroll_cmds.py</strong><dd><code>Improved descriptions and simplified display names in scroll_cmds </code><br><code>actions.</code></dd></summary>
<hr>
      
composio/local_tools/local_workspace/cmd_manager/actions/scroll_cmds.py

<li>Updated descriptions for <code>ScrollDown</code> and <code>ScrollUp</code> actions.<br> <li> Simplified <code>_display_name</code> attributes.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-1f17f84cb9a6ce2f1fc1e06eced9c5ad561095f31135159a25853246961fa99d">+5/-5</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>search_cmds.py</strong><dd><code>Improved descriptions and docstrings in search_cmds actions.</code></dd></summary>
<hr>
      
composio/local_tools/local_workspace/cmd_manager/actions/search_cmds.py

<li>Updated descriptions for <code>SearchDirRequest</code> and <code>FindFileRequest</code> fields.<br> <li> Simplified <code>_display_name</code> attributes.<br> <li> Enhanced docstrings for <code>SearchDirCmd</code>, <code>SearchFileCmd</code>, and <code>FindFileCmd</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-d621598b0c59d9f67451b4cb560e2790e41bbf481519104fba2e488560cff1db">+12/-9</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>cmd_manager_tool.py</strong><dd><code>Commented out SetCursors action in cmd_manager_tool.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/local_workspace/cmd_manager/cmd_manager_tool.py

- Commented out `SetCursors` action from the list of actions.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-6c76fb62718b45cc7010dac52689f7fe9f2750adf1793215232007baa7b73303">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>try-swe.py</strong><dd><code>Refactored try-swe.py for improved readability and functionality.</code></dd></summary>
<hr>
      
examples/swe/try-swe.py

<li>Refactored the script to improve readability and functionality.<br> <li> Added assertions for environment variables.<br> <li> Enhanced agent and task initialization.<br>

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-1b1ee45f402a2c8f917fdd3dd5fe67374d60a3aa65f6b7e3513cf64effefb7d7">+74/-31</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>task_config.yaml</strong><dd><code>Simplified and updated task configuration in task_config.yaml.</code></dd></summary>
<hr>
      
examples/swe/task_config.yaml

- Simplified and updated the task configuration.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-67a5cbef4c04e72dc215d92b2cb2fd810d2aa77206193c707380107b62cf712f">+25/-65</a>&nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Miscellaneous
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>command_runner_config.py</strong><dd><code>Removed command_runner_config.py file.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/local_workspace/config/command_runner_config.py

- Removed the entire `command_runner_config.py` file.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-91f2de21164975eaf1581254f378871f262fccdad009403b16e7f8a88b5c0790">+0/-85</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Documentation
</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>history_keeper_tool.py</strong><dd><code>Updated docstring for HistoryKeeper class.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/local_workspace/history_keeper/history_keeper_tool.py

- Updated docstring for `HistoryKeeper` class.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-7fe858d6c4db1b5674a6eec5f99ee37219ea1c77a39fc06e5185053cde5026e3">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>create_workspace.py</strong><dd><code>Simplified docstring for CreateWorkspaceAction.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/local_workspace/workspace/actions/create_workspace.py

- Simplified docstring for `CreateWorkspaceAction`.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-c6b203e8914f2ebe4ae578f0ad9b4c676eb1a8b3bf6c1276cd6d9db6ba4a55cb">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>setup_github_repo.py</strong><dd><code>Simplified docstring for SetupGithubRepo class.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/local_workspace/workspace/actions/setup_github_repo.py

- Simplified docstring for `SetupGithubRepo` class.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-987916ffcf3c53004176a2677b62ea36194d6d9d5d7c4fd4252ffc7e45099f31">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>workspace_setup.py</strong><dd><code>Simplified docstring for SetupWorkspace class.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/local_workspace/workspace/actions/workspace_setup.py

- Simplified docstring for `SetupWorkspace` class.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-413eb17995e49a8f5384eb65b103577395e117cc5c90a40e240e5d5edfe2fd77">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>workspace_status.py</strong><dd><code>Simplified docstring for WorkspaceStatus class.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
composio/local_tools/local_workspace/workspace/actions/workspace_status.py

- Simplified docstring for `WorkspaceStatus` class.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-090012ba87eb81e7897048955457046a53320897d5101167c2424818314d2cf6">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Formatting
</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>crewai_demo.py</strong><dd><code>Removed unnecessary blank line in crewai_demo.py.</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>
      
plugins/crew_ai/crewai_demo.py

- Removed an unnecessary blank line.

</details>
    

  </td>
  <td><a href="https://github.com/SamparkAI/composio/pull/64/files#diff-9b970c938f315bcfb6e9b47bbcf97182d86e91efde665514f791cb6a6ad0d1e9">+0/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: composio/local_tools/local_workspace/config/default.yaml [hunk 1]
```diff
@@ -1,96 +1,5 @@
...
-    state() {
-      local working_dir="$PWD";
-      if [ -z $CURRENT_FILE ]; then
-          echo '{"open_file": "n/a", "working_dir": "'$working_dir'"}';
-      else
-          echo '{"open_file": "'$(realpath $CURRENT_FILE)'", "working_dir": "'$working_dir'"}';
-      fi
-    };
-parse_function: ThoughtActionParser
+system_template:
```
Comment (utkarsh-dixit) [inline composio/local_tools/local_workspace/config/default.yaml:1]: Just commenting for notice, this is intended ?

File: composio/local_tools/local_workspace/config/commands/README.md [hunk 2]
```diff
@@ -1,50 +0,0 @@
-# Command Configuration
```
Comment (utkarsh-dixit) [inline composio/local_tools/local_workspace/config/commands/README.md:1]: Also is this README.md helpful to someone who wants to create a new command action? If yes, let's keep it

File: composio/local_tools/local_workspace/config/default.yaml (hunk 1 above)
Comment (sohamganatra) [reply composio/local_tools/local_workspace/config/default.yaml:1]: Ha for some random reason, it used to fail without this. Still a lot of cleanup is pending, will delete this in next one.

File: composio/local_tools/local_workspace/config/commands/README.md (hunk 2 above)
Comment (sohamganatra) [reply composio/local_tools/local_workspace/config/commands/README.md:1]: Not his was something random which is not needed.

File: composio/local_tools/local_workspace/workspace/actions/create_workspace.py [hunk 3]
```diff
@@ -1,57 +1,211 @@
...
-        self,
-        workspace_factory: WorkspaceManagerFactory,
-        history_processor: HistoryProcessor,
-    ):
-        self.workspace_factory = workspace_factory
-        self.history_processor = history_processor
+    def set_env_variables(self):
+        commands_to_execute = (
+            [self.config.state_command.code]
+            + ["pip install flake8"]
```
Comment (utkarsh-dixit) [inline composio/local_tools/local_workspace/workspace/actions/create_workspace.py:111]: is flake8 installed in workspace for any specific purpose? The create workspace can be faster if we remove it

File: langchain_math.py [hunk 4]
```diff
@@ -1,17 +1,18 @@
 # Initialise imports
-from langchain.agents import create_openai_functions_agent, AgentExecutor
 from langchain import hub
+from langchain.agents import AgentExecutor, create_openai_functions_agent
 from langchain_openai import ChatOpenAI
```
Comment (utkarsh-dixit) [inline langchain_math.py:4]: This seems like an example/sample code. We should move this to `examples/` instead of at the root folder

**Review by utkarsh-dixit — APPROVED** [review]

**Pull request merged**